src/components/FolderModal.jsx

```diff
@@ -13,23 +21,49 @@ import { errorToast } from '../utils/toasts';
 // axios settings
 axios.defaults.withCredentials = true
 
-const FolderModal = ({ displayTitle, displayText, onClose, category, siteName, isCollection }) => {
-  const [newCategoryName, setNewCategoryName] = useState(category)
-
-  const folderNameChangeHandler = (event) => {
-    const { value } = event.target
-    setNewCategoryName(value)
+const selectRenameApiCall = (isCollection, siteName, folderOrCategoryName, subfolderName, newDirectoryName) => {
+  if (isCollection && !subfolderName) {
+    const params = {
+      siteName,
+      folderName: folderOrCategoryName,
+      newFolderName: newDirectoryName,
+    }
+    return renameFolder(params)
   }
 
-  const saveHandler = async () => {
-    try {
-      await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/${isCollection ? 'collections' : 'resources'}/${category}/rename/${newCategoryName}`)
-      // Refresh page
-      window.location.reload();
-    } catch (err) {
-      errorToast(`There was a problem trying to rename this folder. ${DEFAULT_RETRY_MSG}`)
-      console.log(err);
+  if (isCollection && subfolderName) {
+    const params = {
+      siteName,
+      folderName: folderOrCategoryName,
+      subfolderName,
+      newSubfolderName: newDirectoryName,
     }
+    return renameSubfolder(params)
+  }
+
+  const params = {
+    siteName,
+    categoryName: folderOrCategoryName,
+    newCategoryName: newDirectoryName,
+  }
+  return renameResourceCategory(params)
+}
```

Comment:
We should slugify the new folder/subfolder/category names

Reply:
resolved with [50b0b62](https://github.com/isomerpages/isomercms-frontend/pull/383/commits/50b0b625cdad0961e4c725471f97bd5f56e1136a)